src/main/java/net/sf/rails/ui/swing/AutoLoadPoller.java

```diff
@@ -13,91 +13,102 @@
     public static final int OFF = 0;
     public static final int ON = 1;
     public static final int SUSPENDED = 2;
 
-    protected static Logger log =
-        LoggerFactory.getLogger(AutoLoadPoller.class);
+    protected static Logger log = LoggerFactory.getLogger(AutoLoadPoller.class);
 
     public AutoLoadPoller (GameUIManager guiMgr, String saveDirectory, String savePrefix, String ownPostfix,
             int status, int pollingInterval) {
- 
         this.guiMgr = guiMgr;
         this.saveDirectory = saveDirectory;
         this.savePrefix = savePrefix;
         this.ownPostfix = ownPostfix;
         this.pollingStatus = status;
         this.pollingInterval = pollingInterval;
-        
+
         lastSavedFilenameFilepath = saveDirectory + "/" + savePrefix + ".last_rails";
-        
-        log.debug("Poller own postfix: "+ownPostfix);
-        log.debug("Poller last-filename path: "+lastSavedFilenameFilepath);
-        
+
+        log.debug("Poller own postfix: {}", ownPostfix);
+        log.debug("Poller last-filename path: {}", lastSavedFilenameFilepath);
     }
 
     @Override
     public void run () {
-
         log.info ("AutoLoadPoller started");
 
+        int currentPollInterval = 1;
         int secs, sleepTime;
-        String currentFilename;
 
         for (;;) {
+            secs = Calendar.getInstance().get(Calendar.SECOND);
+            try {
+                sleepTime = 1000 * (currentPollInterval - secs%currentPollInterval);
+                Thread.sleep(sleepTime);
+            } catch (InterruptedException e) {
+                continue;
+            }
 
-            log.debug ("Polling cycle, status="+pollingStatus+" active="+pollingActive);
-            // Process
             if (pollingActive && pollingStatus == ON) {
-                log.debug("Polling...");
                 try {
                     BufferedReader in = new BufferedReader (new FileReader (lastSavedFilenameFilepath));
-                    currentFilename = in.readLine();
+                    String currentFilename = in.readLine();
+                    String fileSize = in.readLine();
                     in.close();
-                    log.debug("Read filename "+currentFilename+"; last saved filename "+lastSavedFilename);
-                    
+                    log.debug("Read filename {}; last saved filename {}", currentFilename, lastSavedFilename);
+
+                    File currFile = new File(saveDirectory+"/"+currentFilename);
+                    if ( ! currFile.exists() ) {
+                        log.debug("Saved file {} missing, waiting", currFile);
+                        currentPollInterval = 1;
+                        continue;
+                    }
+
                     if (!lastSavedFilename.equals(currentFilename)) {
+                        if ( fileSize != null ) {
+                            long fileSizeNum = Long.parseLong(fileSize);
+                            if ( currFile.getTotalSpace() != fileSizeNum ) {
```

Comment:
Maybe shorter: `if (fileSize != null && Long.parseLong(fileSize) != currFile.getTotalSpace())`

This would allow us to remove the inner loop

Reply:
Yes and makes the code harder to read and for the brain to parse given the 3 conditions in the `if ()`. In most cases I prefer the code as written as it is not only easier to read and understand but also because collapsing it down to a single line doesn't buy us anything-- the only time I tend to allow for harder to read code is if there is a strong benefit such as some type of optimization or performance -- but even those are questionable changes with the compilers of today. You would also then have a second hit on getting the file sizes for the logging statement a couple of lines down that you don't have.